dashboard/app/models/user.rb

```diff
@@ -396,6 +384,20 @@ def sanitize_race_data
     end
   end
 
+  def fix_by_user_type
+    if student?
+      self.email = ''
+      self.full_address = nil
+      self.school_info = nil
+    end
+
+    # As we want teachers to explicitly accept our Terms of Service, when the user_type is changing
+    # without an explicit acceptance, we clear the version accepted.
+    if teacher? && user_type_changed? && !terms_of_service_version_changed?
```

Comment:
Is there a better way, @ewjordan, to skip resetting `terms_of_service_version` on account creation? Without the `!terms_of_service_version_changed?` condition, this executes for, e.g., `create :teacher`.